modules/stadeniolib/main.nf

```diff
@@ -0,0 +1,41 @@
+process STADENIOLIB {
+    tag "$meta.id"
+    label 'process_medium'
+
+    conda (params.enable_conda ? "bioconda::staden_io_lib=1.14.14" : null)
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/staden_io_lib:1.14.14--h0d9da7e_3' :
+        'quay.io/biocontainers/staden_io_lib:1.14.14--h0d9da7e_3' }"
+
+    input:
+    tuple val(meta), path(bam)
+    path(fasta)
+    path(fai)
+
+    output:
+    tuple val(meta), path("*.cram") ,emit: cram
+    path "versions.yml"             ,emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
+
+    """
+    scramble \
+        $args \
+        -I bam \
+        -O cram \
+        -r ${fasta} \
+        -t $task.cpus \
+        ${bam} \
+        ${prefix}.cram
+
+    cat <<-END_VERSIONS > versions.yml
+    "${task.process}":
+        stadeniolib: \$(echo \$(scramble -h | head -n 1 |sed 's/^.*version //'))
```

Comment:
Is it correct that the name of the module is staden_io_lib if the tool is called `scramble` :thinking:

Reply:
Well, scramble is actually a subtool from io_lib, so I suppose I should change that.

Reply:
I did this to avoid name collisions with another tool named `scramble`

Reply:
Ok, in that case I would suggest restructing as : staden_io_lib/scramble